tests/app/main/views/test_suppliers.py

```diff
@@ -368,6 +368,181 @@ def test_should_show_separate_tables_for_frameworks_if_supplier_has_service_on_f
+
+        document = html.fromstring(response.get_data(as_text=True))
+        expected_banner_message = "Are you sure you want to remove Supplier Name's 'G-Cloud 8' services?"
+        banner_message = document.xpath('//p[@class="banner-message"]//text()')[0].strip()
+        assert banner_message == expected_banner_message
+
+
+@mock.patch('app.main.views.suppliers.data_api_client')
+class TestDisableSupplierServicesView(LoggedInApplicationTest):
+
+    @pytest.mark.parametrize('url_suffix', ['', '?remove=', '?foo=bar'])
+    def test_400_if_no_framework_provided(self, data_api_client, url_suffix):
+
+        data_api_client.get_supplier.return_value = self.load_example_listing('supplier_response')
+        data_api_client.find_services.return_value = self.load_example_listing('services_response')
+
+        response = self.client.post('/admin/suppliers/1000/services{}'.format(url_suffix))
+
+        assert response.status_code == 400
+
+    def test_400_if_supplier_has_no_service_on_framework(self, data_api_client):
+        framework = 'g-cloud-8'
+        data_api_client.get_supplier.return_value = self.load_example_listing('supplier_response')
+        data_api_client.find_services.return_value = {'services': []}
+
+        response = self.client.post('/admin/suppliers/1000/services?remove={}'.format(framework))
+
+        assert response.status_code == 400
+
+    def test_disables_service(self, data_api_client):
+        framework = 'g-cloud-8'
+        data_api_client.get_supplier.return_value = self.load_example_listing('supplier_response')
+        data_api_client.find_services.return_value = self.load_example_listing('services_response')
+
+        response = self.client.post('/admin/suppliers/1000/services?remove={}'.format(framework))
+
+        assert response.status_code == 302
+        assert data_api_client.update_service_status.called_once_with(
+            '5687123785023488', 'disabled', 'test@example.com'
+        )
+
+    def test_disables_multiple_services(self, data_api_client):
+        framework = 'g-cloud-8'
+        service_1 = self.load_example_listing('services_response')['services'][0]
+        service_2 = service_1.copy()
+        service_2['id'] = '5687123785023489'
+        service_3 = service_1.copy()
+        service_3['id'] = '5687123785023490'
+
+        data_api_client.get_supplier.return_value = self.load_example_listing('supplier_response')
+        data_api_client.find_services.return_value = {'services': [service_1, service_2, service_3]}
+
+        response = self.client.post('/admin/suppliers/1000/services?remove={}'.format(framework))
+
+        assert response.status_code == 302
+        assert data_api_client.update_service_status.call_args_list == [
+            mock.call('5687123785023488', 'disabled', 'test@example.com'),
+            mock.call('5687123785023489', 'disabled', 'test@example.com'),
+            mock.call('5687123785023490', 'disabled', 'test@example.com'),
+        ]
```

Comment:
Is it worth asserting here that the services are all now displayed as 'removed' on the page, and that the 'remove services' link is now hidden? Or is that covered by the earlier tests?

Reply:
That's in `test_no_remove_all_services_link_if_supplier_service_not_published` on the get route 👍

Reply:
Cool, as long as it's covered somewhere.